src/sorcha/utilities/sorchaConfigs.py

```diff
@@ -0,0 +1,1272 @@
+
+    ephemerides_type: str = ""
+    """Simulation used for ephemeris input."""
+
+    eph_format: str = ""
+    """Format for ephemeris simulation input file."""
+
+    size_serial_chunk: int = 0
+    """Sorcha chunk size."""
+
+    aux_format: str = ""
+    """Format for the auxiliary input files."""
+
+    pointing_sql_query: str = ""
+    """SQL query for extracting data from pointing database."""
+
+    def __post_init__(self):
+        """Automagically validates the input configs after initialisation."""
+        self._validate_input_configs()
+
+    def _validate_input_configs(self):
+        """
+        Validates the input config attributes after initialisation.
+
+        Parameters
+        -----------
+        None.
+
+        Returns
+        ----------
+        None
+        """
+        # make sure all the mandatory keys have been populated.
+        check_key_exists(self.ephemerides_type, "ephemerides_type")
+        check_key_exists(self.eph_format, "eph_format")
+        check_key_exists(self.size_serial_chunk, "size_serial_chunk")
+        check_key_exists(self.aux_format, "aux_format")
+        check_key_exists(self.pointing_sql_query, "pointing_sql_query")
+
+        # some additional checks to make sure they all make sense!
+        check_value_in_list(self.ephemerides_type, ["ar", "external"], "ephemerides_type")
+        check_value_in_list(self.eph_format, ["csv", "whitespace", "hdf5"], "eph_format")
+        check_value_in_list(self.aux_format, ["comma", "whitespace", "csv"], "aux_format")
+        self.size_serial_chunk = cast_as_int(self.size_serial_chunk, "size_serial_chunk")
+
+
+@dataclass
+class simulationConfigs:
+    """Data class for holding SIMULATION section configuration file keys and validating them"""
+
+    ar_ang_fov: float = 0.0
+    """the field of view of our search field, in degrees"""
+
+    ar_fov_buffer: float = 0.0
+    """the buffer zone around the field of view we want to include, in degrees"""
+
+    ar_picket: float = 0.0
+    """imprecise discretization of time that allows us to move progress our simulations forward without getting too granular when we don't have to. the unit is number of days."""
+
+    ar_obs_code: str = ""
+    """the obscode is the MPC observatory code for the provided telescope."""
+
+    ar_healpix_order: int = 0
+    """the order of healpix which we will use for the healpy portions of the code."""
+
+    _ephemerides_type: str = ""
+    """Simulation used for ephemeris input."""
+
+    def __post_init__(self):
+        """Automagically validates the simulation configs after initialisation."""
+        self._validate_simulation_configs()
+
+    def _validate_simulation_configs(self):
+        """
+        Validates the simulation config attributes after initialisation.
+
+        Parameters
+        -----------
+        None.
+
+        Returns
+        ----------
+        None
+        """
+        # make sure all the mandatory keys have been populated.
```

Comment:
self._ephmerides_type is read into simultionConfigs from the inputConfig class, so a check has already been made to ensure that it is in that list. Should I create the check anyway for when simulationConfigs is tested independently? 

I put an underscore in front of it to indicate that it should only be used for checks in simulationConfigs because ephemerides_type is called from the inputConfigs class for the rest of the code.

Reply:
Oh, I see what you've done! Nah, you're good, that's good practice.